### ISSUE TITLE
Error Setting up entry Simple Wyze Vac for simple_wyze_vac

### ISSUE THREAD

**Issue by dwdonnelly** [romedtino/simple-wyze-vac#80]
**Describe the bug**
Error Setting up entry Simple Wyze Vac for simple_wyze_vac

**To Reproduce**
Each time the vacuum automation is run.  This automation has been active for over 1 year with no errors.

Attempted to reload integration.  Reload stated success, but the error is still present.
Restarted HA, but the error is still present.
Obtained new API/KEY and Base32 Secret for 2FA and applied during re-install of integration.
Reload stated success.
Restarted HA, but the error is still present.

**Expected behavior**
The vacuum automation to run as before.

**Screenshots**
`Logger: homeassistant.config_entries
Source: config_entries.py:586
First occurred: July 3, 2024 at 1:17:41 PM (5 occurrences)
Last logged: 10:28:13 AM

Error setting up entry Simple Wyze Vac for simple_wyze_vac
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 586, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/simple_wyze_vac/__init__.py", line 52, in async_setup_entry
    client = await hass.async_add_executor_job(Client, None, None, username, password, key_id, api_key, totp)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/wyze_sdk/api/client.py", line 72, in __init__
    self.login()
  File "/usr/local/lib/python3.12/site-packages/wyze_sdk/api/client.py", line 184, in login
    response = self._auth_client().user_login(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/wyze_sdk/service/auth_service.py", line 88, in user_login
    response = self.api_call(
               ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/wyze_sdk/service/auth_service.py", line 58, in api_call
    return super().api_call(
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/wyze_sdk/service/base.py", line 347, in api_call
    return super().api_call(
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/wyze_sdk/service/base.py", line 204, in api_call
    return self.do_post(url=api_url, headers=headers, payload=json, params=params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/wyze_sdk/service/base.py", line 132, in do_post
    return self._do_request(client, req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/wyze_sdk/service/base.py", line 106, in _do_request
    raise err
  File "/usr/local/lib/python3.12/site-packages/wyze_sdk/service/base.py", line 93, in _do_request
    response.raise_for_status()
  File "/usr/local/lib/python3.12/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/api/user/login`

**Simple Wyze Vac Version**
- best I can tell - it is 1.9.0 

**Wyze Vacuum Firmware Version**
-1.6.202

**Home Assistant Version**
-2024.7.0

**Additional context**
None

**Comment by dwdonnelly** [comment]
@romedtino Just following up to ee if there is any update on this issue?

**Comment by romedtino** [comment]
Hm, Your traceback indicate a credential/login issue especially with the Error 400 which indicate the server was reached but it was given bad data. Given that the traceback calls the API login, it seems related to an issue with your account?

**Comment by dwdonnelly** [comment]
@romedtino In working on the problem, I got a new API key, Key ID, and reused my existing 32bit code. I reinstalled the integration.  Went through the setup process with no errors.

Then I tested the automation to run the two wyze vacuums and got this error from the 'trace':
`Executed: July 11, 2024 at 4:48:15 PM
Error: The request to the Wyze API failed. The server responded with: {'code': 3000, 'current': 0, 'data': None, 'hash': '1', 'instance_id': 'bc8a782462df12ed189a18754270bb74', 'isShowToast': 0, 'message': 'Device is offline', 'toastMsg': None, 'total': 0, 'version': 1}
Result:
params:
  domain: vacuum
  service: send_command
  service_data:
    params:
      rooms:
        - Hallway
        - Kitchen
    command: sweep_rooms
    entity_id:
      - vacuum.irobot
  target:
    entity_id:
      - vacuum.irobot
running_script: false`


From the log I get:
`This error originated from a custom integration.

Logger: custom_components.simple_wyze_vac.vacuum
Source: custom_components/simple_wyze_vac/vacuum.py:280
integration: Simple Wyze Vac (documentation)
First occurred: July 11, 2024 at 4:48:15 PM (1 occurrences)
Last logged: July 11, 2024 at 4:48:15 PM

Received WyzeApiError`

I am guessing I need to get a new 32bit code..or a whole new setup: API key, Key ID, and 32bit code.  But that seems odd.

Thoughts?

**Comment by dwdonnelly** [comment]
@romedtino   I think fundamentally you are correct, in that it is a problem outside of HA and this integration.  Based on some further testing, it has become apparent that one of my two wyze vacuums has begun to randomly drop offline.  This is causing the automation to stop working.

Now, maybe that in itself is an issue.  It would seem if the wyze vacuum, in this case, dropped offline, the automation would recognize this issue, and proceed with the rest of the actions within the automation. 

Pending your feedback, it would seem correct to close this particular issue.

Standing by.

**Comment by dwdonnelly** [comment]
@romedtino Well, I am confused on this issue.  Several days ago, this one wyze vacuum's lack of wifi connectivity seemed to just solve itself.  Meaning, following your suggestion, I contacted Wyze, performed all their troubleshooting recommendations, provided logs, etc, but collectively we could find no issue.  And nothing we did corrected the problem - the constant dropping of the wifi connection.  Then, a couple of days after the troubleshooting efforts and contemplating ordering a replacement, the vacuum suddenly connected to the house wifi.  I ran several tests of its operation over several days and it is operating just fine, via the Wyze app and HA & your integration.

  So, at this point, I am marking the problem up to Sun Spots or some gremlin who got tired and moved to another house.  And yes, it could be the start of a random problem indicating the wifi system in the vacuum is slowly failing.   Have to take a wait and see approach.

At any rate, thank you for the assist, and I am closing this issue.
